### PR TITLE
Track 103 - Extends seed script with random grades

### PR DESCRIPTION
**Pull request by tomca32**

### CODE REVIEW

File: db/seed_grades.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  seed_student_grades student, lesson
+end
+
+def mark_absent(student, lesson)
+  lesson.mark_student_as_absent student
+end
+
+def seed_student_grades(student, lesson)
+  lesson.subject.skills.each do |skill|
```
Comment (j16r) [inline db/seed_grades.rb]: Might be effective to use `build` here then save the student afterwards which should save associated records in one hit.

Comment (tomca32) [inline db/seed_grades.rb]: good point

Comment (tomca32) [inline db/seed_grades.rb]: This is what I did:

```ruby
def seed_student_grades(student, lesson)
  lesson.subject.skills.each do |skill|
    student.grades.build lesson: lesson, student: student, grade_descriptor: skill.grade_descriptors.sample
  end
  student.save
end
```
For some interesting reason this makes it around 20% slower when seeding ~10k records.

Takes ~27s on average while the old solution with using create in the loop takes ~22s on average.

Comment (tomca32) [inline db/seed_grades.rb]: Actually, doing:

```ruby
def seed_student_grades(student, lesson)
  lesson.subject.skills.each do |skill|
    Grade.create lesson: lesson, student: student, grade_descriptor: skill.grade_descriptors.sample
  end
end
```

Seems to be the fastest. ~20s on average on my machine

Comment (j16r) [inline db/seed_grades.rb]: Huh, interesting - I guess it's not quite doing it as efficiently as I had hoped. Is there no bulk creation mechanism?

Comment (tomca32) [inline db/seed_grades.rb]: Doesn't seem like it. Best I could find is to wrap it in a transaction and that does help:

```ruby
def seed_student_grades(student, lesson)
  Grade.transaction do
    lesson.subject.skills.each do |skill|
      Grade.create lesson: lesson, student: student, grade_descriptor: skill.grade_descriptors.sample
    end
  end
end
```
~17s on average.

**Comment by j16r** [comment]
good enough, ship it!

**Pull request merged**